torch/csrc/autograd/engine.h

```diff
@@ -45,6 +147,24 @@ struct TORCH_API Engine {
       bool keep_graph,
       bool create_graph,
       const edge_list& outputs = {});
+
+  // Given a pre-populated GraphTask and GraphRoot, computes the backward pass
+  // for the graph. This API should only be used by internal autograd specific
+  // machinery and shouldn't be exposed to users in anyway.
+  variable_list execute_with_graph_task(
+      GraphTask& graph_task,
+      std::shared_ptr<Node> graph_root);
+
+  // Enqueues a task for execution on the CPU thread. This method assumes that
+  // the appropriate GraphTask has already been initialized appropriately.
+  // Another important part is that this does not increment 'outstanding_tasks_'
+  // in the appropriate GraphTask. It is assumed we've already done this before
+  // hand for this task. This is useful in the distributed autograd case where
+  // we need to increment 'outstanding_tasks_' first to indicate the local
+  // autograd engine needs to wait for this task, but the task might actually
+  // be received later over the network for execution.
+  void enqueue_on_cpu(NodeTask task);
```

Comment:
The name `enqueue_on_cpu` is as clear as mud; if I saw that name by itself, I wouldn't have guessed that it doesn't increment outstanding tasks.

I think the naming here should be more in the spirit of the "blocking" natures of these tasks. For example, one mental model that makes sense to me is that we have two "classes" of tasks: "ready to execute" tasks and "blocked" tasks.  `outstanding_tasks_` counts both of them. But for blocked tasks, we must manually unblock them... via this API!